mysql-test/suite/tokudb.rpl/t/rpl_tokudb_rfr_partition_table.test

```diff
@@ -0,0 +1,70 @@
+# test tokudb read free replication feature with partition table
+
+--source include/have_debug.inc
+--source include/have_tokudb.inc
+--source include/have_binlog_format_row.inc
+--source include/master-slave.inc
+
+call mtr.add_suppression(".*read free replication is disabled for TokuDB table.*continue with rows lookup");
+
+connection master;
+
+# partition table with explicit PK
+CREATE TABLE t1 (id int(11) NOT NULL, pid int(11), PRIMARY KEY (id)) ENGINE=TokuDB
+PARTITION BY RANGE (id)
+(PARTITION p_1 VALUES LESS THAN (10) ENGINE = TokuDB,
+ PARTITION p_2 VALUES LESS THAN (20) ENGINE = TokuDB,
+ PARTITION p_all VALUES LESS THAN MAXVALUE ENGINE = TokuDB);
+
+insert into t1 values (1, 1), (2, 2), (3, 3), (11, 11), (12, 12), (13, 13);
+
+# partition table without explicit PK
+CREATE TABLE t2 (id int(11) NOT NULL, pid int(11), key idx_1(id)) ENGINE=TokuDB
+PARTITION BY RANGE (id)
+(PARTITION p_1 VALUES LESS THAN (10) ENGINE = TokuDB,
+ PARTITION p_2 VALUES LESS THAN (20) ENGINE = TokuDB,
+ PARTITION p_all VALUES LESS THAN MAXVALUE ENGINE = TokuDB);
+
+insert into t2 values (1, 1), (2, 2), (3, 3), (11, 11), (12, 12), (13, 13);
+
+--sync_slave_with_master
+
+# set tokudb rfr delay
+connection slave;
+set global debug= "+d,tokudb_in_rpl_lookup_rows,tokudb_rpl_do_unique_checks";
```

Comment:
This facility is not multi-threaded, here you are updating its setting while an async background thread - slave SQL thread - might be reading it, which results in crashes like https://bugs.mysql.com/bug.php?id=58754. I don't know if fully-synced SQL thread is safe here or not. Since you are restarting the slave SQL thread here anyway, just move the stop slave sql_thread above the debug setting (and replace both stop/start slave with include/stop|start_slave_sql.inc for synchronisation)